### PR TITLE
Add touches controls

### PR DESCRIPTION
**Pull request by apomelitos**

### CODE REVIEW

File: src/hooks/useHandleTouches.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return startCoords.clientX > endCoords.clientX ? 'LEFT' : 'RIGHT';
+  } else if (absY > MIN_DISTANCE) {
+    return startCoords.clientY > endCoords.clientY ? 'UP' : 'DOWN';
+  }
+
+  return undefined;
+};
+
+export const useHandleTouches = (onSwipe: (direction: Direction) => void) => {
+  const [startCoords, setStartCoords] = useState<Coords | null>(null);
```
Comment (mrlika) [inline src/hooks/useHandleTouches.ts]: `startCoords` doesn't participate in rendering and is not a dependency of a hook. It means that it is better to implement it as `Ref`

**Review by mrlika — CHANGES_REQUESTED** [review]

File: src/components/Game.tsx [hunk 2]
```diff
@@ -84,6 +85,7 @@ export const Game: FC = (): JSX.Element => {
   };
 
   useHandleButtons(updateState);
+  const [onTouchStart, onTouchEnd] = useHandleTouches((dir) => updateState(dir));
```
Comment (mrlika) [inline src/components/Game.tsx]: This line can be simplified to `const [onTouchStart, onTouchEnd] = useHandleTouches(updateState);`

**Review by mrlika — CHANGES_REQUESTED** [review]

File: src/utils/gameUtils.ts [hunk 3]
```diff
@@ -4,6 +4,16 @@ type Position = [number, number];
 
 type MoveStateReturn = [TileMeta[], [TileMeta, TileMeta][], number];
 
+function* createGenerator(start: number) {
+  let current = start;
+  while (true) {
+    yield current++;
+    console.log('Hello');
```
Comment (mrlika) [inline src/utils/gameUtils.ts]: Remove `console` statement

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by TheForsakenSpirit — APPROVED** [review]

File: src/utils/gameUtils.ts (hunk 3 above)
Comment (apomelitos) [reply src/utils/gameUtils.ts]: Already removed in the last commit [961ab3d](https://github.com/apomelitos/2048-typescript/pull/12/commits/961ab3d3cb90127f53aa65638c6d394a6e29b4c4)

**Review by mrlika — APPROVED** [review]

**Pull request merged**